rerun_py/tests/unit/test_datafusion_tables.py

```diff
@@ -162,6 +162,40 @@ def test_df_aggregation(server_instance: tuple[subprocess.Popen[str], DatasetEnt
     assert results[0][1][0] == pa.scalar(50.0, type=pa.float32())
 
 
+def test_component_filtering(server_instance: tuple[subprocess.Popen[str], DatasetEntry]) -> None:
+    """
+    Cover the case where a user specifies a component filter on the client.
+
+    We also support push down filtering to take a `.filter()` on the dataframe gets
+    pushed into the query. Verify these both give the same results and that we don't
+    get any nulls in that column.
+    """
+    (_process, dataset) = server_instance
+
+    component_path = "/obj2:Points3D:positions"
+
+    filter_on_query = (
+        dataset.dataframe_query_view(index="time_1", contents="/**")
+        .filter_is_not_null(component_path)
+        .df()
+        .collect_partitioned()
+    )
+
+    filter_on_dataframe = (
+        dataset.dataframe_query_view(index="time_1", contents="/**")
+        .filter_is_not_null(component_path)
+        .df()
```

Comment:
I assume you meant this to be 
```suggestion
        .df()
        .filter(col(component_path).is_not_null())
```

Reply:
Yes, thank you